src/components/LaunchDetails.js

```diff
@@ -0,0 +1,104 @@
+import { useParams } from "react-router-dom";
+import useFetch from "../customHooks/useFetch";
+import { useHistory } from 'react-router-dom'
+import { useContext } from 'react';
+import ThemeContext from '../context/ThemeContext';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import Grid from '@mui/material/Grid';
+
+const LaunchDetails = () => {
+    const { id } = useParams();
+    const { data: launch, error, isPending } = useFetch('https://api.spacexdata.com/v3/launches/' + id);
```

Comment:
API url should be env variable. Reference: `https://create-react-app.dev/docs/adding-custom-environment-variables/`